### PR TITLE
Add sphinx.ext.githubpages to publish the docs on GitHub Pages

### LINKED ISSUE #2203: configuration to use specified directory name instead of underscore prefixed names as `_static`
For what?
- for github pages (now a days, `.nojekyll` is recommended)
- some other purpose?

Targets:
- _static
- _images
- _downloads
- _sources

Limitations:
- Conflicts with user's directories, however user should know.
- Some extensions write files into `_static` or `_images` directories directory, maybe (ex: diagram or some image generators).

Relates:
- #2202 want to use `static` instead of `_static`
- [sphinxtogithub](https://pypi.python.org/pypi/sphinxtogithub) plugin rename underscore prefixed directories on the `build-finished` event.

### PR DESCRIPTION
**Pull request by tk0miya**
To resolve #2203, I added `sphinx.ext.githubpages` and ask it on quickstart.

### CODE REVIEW

File: doc/ext/githubpages.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. highlight:: rest
```
Comment (birkenfeld) [inline doc/ext/githubpages.rst:1]: Doesn't this have to be added to doc/extensions.rst?

**Comment by birkenfeld** [comment]
Looks good!

File: doc/ext/githubpages.rst (hunk 1 above)
Comment (tk0miya) [inline doc/ext/githubpages.rst:1]: Thanks! I missed it.
Fixed now :-)

**Comment by tk0miya** [comment]
@shimizukawa Does this resolve #2203? If it's true, I'll merge this.

**Comment by shimizukawa** [comment]
@tk0miya #2203 is not only for github pages, however actually there is no reason to rename directory names that prefixed with underscore I think. For now, I think #2203 is fixed by this PR. Thanks for your necessary and sufficient solution!

**Comment by shimizukawa** [comment]
@tk0miya please update CHANGES for this feature ;)

**Comment by tk0miya** [comment]
@shimizukawa Ok, I'll update it after merging.
(I do not include CHANGES intentionally. It often conflicts...)

**Pull request merged**